aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1257,13 +1257,49 @@ static inline Tensor& bmm_out_or_baddbmm_(Tensor& self_or_result, const Tensor&
     at::native::_baddbmm_mkl_(self_or_result, batch1, batch2, beta, alpha);
   } else { // split along batch dimension
     if (is_bmm_out) {
-      for (int64_t b = 0; b < bs; b++) {
-        auto r = self_or_result.select(0, b);
-        addmm_impl_cpu_(r, r, batch1.select(0, b), batch2.select(0, b), 0, 1);
+      /*
+       * Inference mode multithreading is done because various thread local
```

Comment:
Would you mind also opening an issue and documenting what's the new API would do there? :D

Reply:
Also I'd appreciate a big warning here: if someone wants to add another instance of this they should do it properly by using the new API. :D